docs/source/upgrading/4.2-4.3.rst

```diff
@@ -96,3 +96,27 @@ Configuration variables
                 }
             }
 
+
+Installable File System (IFS)
+-----------------------------
+
+Sming now supports multiple filesystems via :component:`IFS`.
+
+See :sample:`Basic_IFS` for a demonstration.
+
+:source:`Core/FileSystem.h` has been modified to use IFS but the API remains largely unchanged, although somewhat expanded.
+Functions are now mainly just wrappers around filing system calls.
+
+A single global IFileSystem instance is used.
+
+SPIFFS
+    All access is now managed via the ``IFS::SPIFFS::FileSystem`` implementation.
+
+    Applications should not use SPIFFS functions directly.
+
+    SPIFFS is now built with ``SPIFFS_OBJ_META_LEN=16`` to store extended attribute information,
+    but existing volumes should still be readable.
```

Comment:
Quote from https://github.com/pellepl/spiffs/blob/9d12e8f47b0e611f277a0de8d39dc9089971ce20/src/default/spiffs_config.h#L155

> // Setting to non-zero value enables metadata-related API but also
> // changes the on-disk format, so the change is not backward-compatible.

If I understand this correctly all existing spiffs that we have up to now will not be readable any more. @mikee47 Can you check if this is the case? If yes - we should update the documentation and also the upgrade document.

Reply:
I tested this with the `Basic_IFS` sample for Host since it gives a detailed file listing, replacing the call to `initFileSystem()` with a `spiffs_mount()`. After a `make flash` I then built the `HttpServer_ConfigNetwork` sample and flashed the resulting `spiff_rom.bin`. This is what we get:

```
#     Name                             Modified               Size       Size (KB)  Orig Size  Attributes Access
----- -------------------------------- ---------------------- ---------- ---------- ---------- ---------- ----------
0     core.js.gz                       2021-02-10T14:58:26Z   30474      29.76      0          .....      aa
1     bootstrap-core.css.gz            2021-02-10T14:58:26Z   15861      15.49      0          .....      aa
2     wifi-sprites.png                 2021-02-10T14:58:26Z   1769       1.73       0          .....      aa
3     .lastModified                    2021-02-10T14:58:26Z   30         0.03       0          .....      aa
4     settings.html                    2021-02-10T14:58:26Z   2535       2.48       0          .....      aa
5     index.html                       2021-02-10T14:58:26Z   6030       5.89       0          .....      aa
```

I then build the same `HttpServer_ConfigNetwork` sample under `develop` branch and flash the resulting `spiff_rom.bin`:

```
#     Name                             Modified               Size       Size (KB)  Orig Size  Attributes Access
----- -------------------------------- ---------------------- ---------- ---------- ---------- ---------- ----------
0     core.js.gz                       1971-08-06T13:09:20Z   30474      29.76      0          .....      -x
1     bootstrap-core.css.gz            2040-03-06T05:54:08Z   15861      15.49      0          .....      -x
2     wifi-sprites.png                 2075-10-19T13:17:52Z   1769       1.73       0          .....      -x
3     .lastModified                    2105-07-27T16:51:12Z   30         0.03       0          CARDM      xx
4     settings.html                    2081-08-24T13:47:44Z   2535       2.48       0          .....      -x
5     index.html                       2088-01-10T18:42:08Z   6030       5.89       0          .....      -x
```

So the answer I think is YES, it still works, but room for improvement perhaps!

Reply:
I'll investigate further to see if there's a reilable way to detect this. Logically I'd expect the unused area (now occupied by 16 bytes of meta) to be 0xFF, but it's not.

Reply:
For info, although file listing is correct file content is not. Fix coming shortly.